app_grim_reefer/src/main.c

```diff
@@ -62,19 +90,168 @@ static int init(void) {
     LOG_ERR("Unable to configure camera enable output pin\n");
     return -1;
   }
+
+  //   if (!gpio_is_ready_dt(&buzzer)) {
+  // LOG_ERR("buzzer pin is not ready\n");
+  // return -1;
+  //   }
+  if (gpio_pin_configure_dt(&cam_enable, GPIO_OUTPUT_ACTIVE) < 0) {
+    LOG_ERR("Unable to configure buzzer output pin\n");
+    return -1;
+  }
+
+  if (!device_is_ready(debug_serial_dev)) {
+    LOG_ERR("CAMERA SERIAL NOT READY\n");
+    return -1;
+  }
+
+  gpio_pin_set_dt(&led1, 0);
+  gpio_pin_set_dt(&led2, 0);
+  gpio_pin_set_dt(&ldo_enable, 0);
+  gpio_pin_set_dt(&cam_enable, 0);
+
+  return 0;
+}
+
+static int sensor_init(void) {
+  const bool lsm6dsl_found = device_is_ready(lsm6dsl_dev);
+  const bool bme280_found = device_is_ready(bme280_dev);
+  const bool flash_found = device_is_ready(flash_dev);
+
+  struct sensor_value odr_attr;
+
+  /* set accel/gyro sampling frequency to 104 Hz */
+  odr_attr.val1 = 1666;
+  odr_attr.val2 = 0;
+
+  if (sensor_attr_set(lsm6dsl_dev, SENSOR_CHAN_ACCEL_XYZ,
+                      SENSOR_ATTR_SAMPLING_FREQUENCY, &odr_attr) < 0) {
+    printk("Cannot set sampling frequency for accelerometer.\n");
+    return 0;
+  }
+
+  if (sensor_attr_set(lsm6dsl_dev, SENSOR_CHAN_GYRO_XYZ,
+                      SENSOR_ATTR_SAMPLING_FREQUENCY, &odr_attr) < 0) {
+    printk("Cannot set sampling frequency for gyro.\n");
+    return 0;
+  }
+
+  const bool all_good = lsm6dsl_found && flash_found && bme280_found;
+  if (!all_good) {
+    LOG_ERR("Error setting up sensor and flash devices");
+    return -1;
+  }
```

Comment:
Not a fan of this. No single failure in hardware should stop everything, unless its mission critical. I guess you can not do stuff if flash was not found. But if we dont find a LSM6DSL, we should still try to log BME280 data for example